grimoiressg/arguments.py

```diff
@@ -0,0 +1,31 @@
+import argparse
+
+
+def parse_arguments_to_initial_context():
+    parser = argparse.ArgumentParser(
+        description='''
+            Grimoire is a minimalistic Static Site Generator.
+            In the simplest case the only argument needed is at least one content file. \
+            The rest of the flags is used to customize the behavior.
+        '''
+    )
+    parser.add_argument("content_file", nargs='+', help="one or more content files")
+    parser.add_argument("-o", "--output", default="./output/", help="the output directory (default: ./output/)")
+    parser.add_argument("-c", "--config", help="the config file to use")
+
+    args, _ = parser.parse_known_args()
+
+    context = {
+        "output_dir": args.output,
+        "config_file": args.config,
+        "filenames": args.content_file
+    }
```

Comment:
Would it perhaps make more sense to define the context as a class so that it is easier to access?

Reply:
I think I would like to keep it dynamic, so custom modules can modify the context freely without running into any type constraints. Do you think it's worth it?

Reply:
It might overcomplicate things with an extra class. So I guess it is fine.